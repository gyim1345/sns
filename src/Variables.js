export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
