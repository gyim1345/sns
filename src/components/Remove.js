import React, { useState } from "react";
import { Redirect } from "react-router-dom";
import { removePostApi } from "../apis/post";
import PropTypes from "prop-types";
import { Global, css, jsx } from "@emotion/core";

function Remove({
  posting,
  currentUser,
  indexOfCommentOnThisPosting,
  setCommentAPI
}) {
  const [removed, setRemoved] = useState(false);

  const onClick = async () => {
    try {
      const response = await removePostApi(
        posting,
        currentUser,
        indexOfCommentOnThisPosting
      );
      if (response.Message !== undefined) return alert(response.Message);
      indexOfCommentOnThisPosting === undefined
        ? setRemoved(response)
        : setCommentAPI(response);
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <>
      <button
        type="button"
        onClick={onClick}
        id="buttonRemove"
        css={[marginLeft0]}
      >
        Remove
      </button>
      {removed && <Redirect to={`/TimeLine/${currentUser}`} />}
    </>
  );
}

const marginLeft0 = css`
  margin-left: 0px;
`;

Remove.propTypes = {
  currentUser: PropTypes.string,
  posting: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  setCommentAPI: PropTypes.elementType,
  indexOfCommentOnThisPosting: PropTypes.number
};

Remove.defaultProps = {
  currentUser: "",
  posting: {},
  isComment: false
};

export default Remove;
