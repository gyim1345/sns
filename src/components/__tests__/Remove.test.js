import React from 'react';
import { mount } from 'enzyme';
import Remove from '../Remove';
import { post, currentUser } from '../../Variables';

describe('<Remove />', () => {
  it('renders <button> and "Remove"', () => {
    const wrapper = mount(<Remove posting={post} currentUser={currentUser} />);

    expect(wrapper.props().posting).toBe(post);
    expect(wrapper.props().currentUser).toBe(currentUser);
    expect(wrapper.text()).toMatch('Remove');
    expect(wrapper.html()).toMatch('<button');
    expect(wrapper).toMatchSnapshot();
  });
});
