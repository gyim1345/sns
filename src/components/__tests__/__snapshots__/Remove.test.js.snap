// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Remove /> renders <button> and "Remove" 1`] = `
<Remove
  currentUser="gibong@gmail.com"
  posting={
    Object {
      "_id": "5e8ec52d9bb4913b5a844423",
      "id": 1,
      "imageUrl": "http://localhost:3000/static/images/1.jpg",
      "like": Array [
        "gibong@gmail.com",
      ],
      "tag": Array [
        "landscape",
      ],
      "title": "posting with id 1",
      "userName": "gibong@gmail.com",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Remove"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="button"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlbW92ZS5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUF5Q3FCIiwiZmlsZSI6IlJlbW92ZS5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCwgeyB1c2VTdGF0ZSB9IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IFJlZGlyZWN0IH0gZnJvbSAncmVhY3Qtcm91dGVyLWRvbSc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcbmltcG9ydCBTd2FsIGZyb20gJ3N3ZWV0YWxlcnQyJztcblxuaW1wb3J0IHsgcmVtb3ZlUG9zdEFwaSB9IGZyb20gJy4uL2FwaXMvcG9zdCc7XG5cbmZ1bmN0aW9uIFJlbW92ZSh7IHBvc3RpbmcsIHNldFBvc3RpbmcsIHBvc3RpbmdBbGwgfSkge1xuICBjb25zdCBvbkNsaWNrID0gYXN5bmMgKCkgPT4ge1xuICAgIHRyeSB7XG4gICAgICBjb25zdCByZXNwb25zZSA9IGF3YWl0IHJlbW92ZVBvc3RBcGkocG9zdGluZyk7XG4gICAgICBpZiAocmVzcG9uc2UuTWVzc2FnZSAhPT0gdW5kZWZpbmVkKVxuICAgICAgICByZXR1cm4gU3dhbC5maXJlKHtcbiAgICAgICAgICBpY29uOiAnZXJyb3InLFxuICAgICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgICAgdGV4dDogYCR7cmVzcG9uc2UuTWVzc2FnZX1gXG4gICAgICAgIH0pO1xuICAgICAgc2V0UG9zdGluZyhwb3N0aW5nQWxsLmZpbHRlcih4ID0+IHggIT09IHBvc3RpbmcpKTtcbiAgICB9IGNhdGNoIChlKSB7XG4gICAgICBTd2FsLmZpcmUoe1xuICAgICAgICBpY29uOiAnZXJyb3InLFxuICAgICAgICB0aXRsZTogJ09vcHMuLi4nLFxuICAgICAgICB0ZXh0OiAnSW50ZXJuYWwgRXJyb3InXG4gICAgICB9KTtcbiAgICB9XG4gIH07XG5cbiAgcmV0dXJuIChcbiAgICA8PlxuICAgICAgPGJ1dHRvblxuICAgICAgICB0eXBlPVwiYnV0dG9uXCJcbiAgICAgICAgb25DbGljaz17b25DbGlja31cbiAgICAgICAgaWQ9XCJidXR0b25SZW1vdmVcIlxuICAgICAgICBjc3M9e1tidXR0b25Dc3NdfVxuICAgICAgPlxuICAgICAgICBSZW1vdmVcbiAgICAgIDwvYnV0dG9uPlxuICAgIDwvPlxuICApO1xufVxuXG5jb25zdCBidXR0b25Dc3MgPSBjc3NgXG4gIGJvcmRlcjogMXB4IHNvbGlkIHJnYmEodmFyKC0tZDBiLCAyMTksIDIxOSwgMjE5KSwgMSk7XG4gIGNvbG9yOiByZ2JhKHZhcigtLWYwNywgMzgsIDM4LCAzOCksIDEpO1xuICBiYWNrZ3JvdW5kLWNvbG9yOiB3aGl0ZTtcbiAgd2lkdGg6IDMwMHB4O1xuICBoZWlnaHQ6IDQ4cHg7XG4gIGZvbnQtc2l6ZTogMTRweDtcbiAgZm9udC13ZWlnaHQ6IGJvbGQ7XG5gO1xuXG5jb25zdCBtYXJnaW5MZWZ0MCA9IGNzc2BcbiAgbWFyZ2luLWxlZnQ6IDBweDtcbmA7XG5cbi8vIFJlbW92ZS5wcm9wVHlwZXMgPSB7XG4vLyAgIGN1cnJlbnRVc2VyOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICBwb3N0aW5nOiBQcm9wVHlwZXMub25lT2ZUeXBlKFtQcm9wVHlwZXMub2JqZWN0LCBQcm9wVHlwZXMuYXJyYXldKSxcbi8vICAgc2V0Q29tbWVudEFQSTogUHJvcFR5cGVzLmVsZW1lbnRUeXBlLFxuLy8gICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3Rpbmc6IFByb3BUeXBlcy5udW1iZXJcbi8vIH07XG5cbi8vIFJlbW92ZS5kZWZhdWx0UHJvcHMgPSB7XG4vLyAgIGN1cnJlbnRVc2VyOiBcIlwiLFxuLy8gICBwb3N0aW5nOiB7fSxcbi8vICAgaXNDb21tZW50OiBmYWxzZVxuLy8gfTtcblxuZXhwb3J0IGRlZmF1bHQgUmVtb3ZlO1xuIl19 */",
        "name": "e09scu-buttonCss-Remove",
        "next": undefined,
        "styles": "border:1px solid rgba(var(--d0b,219,219,219),1);color:rgba(var(--f07,38,38,38),1);background-color:white;width:300px;height:48px;font-size:14px;font-weight:bold;;label:buttonCss;;;label:Remove;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlbW92ZS5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFpQ1EiLCJmaWxlIjoiUmVtb3ZlLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0LCB7IHVzZVN0YXRlIH0gZnJvbSAncmVhY3QnO1xuaW1wb3J0IHsgUmVkaXJlY3QgfSBmcm9tICdyZWFjdC1yb3V0ZXItZG9tJztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuaW1wb3J0IFN3YWwgZnJvbSAnc3dlZXRhbGVydDInO1xuXG5pbXBvcnQgeyByZW1vdmVQb3N0QXBpIH0gZnJvbSAnLi4vYXBpcy9wb3N0JztcblxuZnVuY3Rpb24gUmVtb3ZlKHsgcG9zdGluZywgc2V0UG9zdGluZywgcG9zdGluZ0FsbCB9KSB7XG4gIGNvbnN0IG9uQ2xpY2sgPSBhc3luYyAoKSA9PiB7XG4gICAgdHJ5IHtcbiAgICAgIGNvbnN0IHJlc3BvbnNlID0gYXdhaXQgcmVtb3ZlUG9zdEFwaShwb3N0aW5nKTtcbiAgICAgIGlmIChyZXNwb25zZS5NZXNzYWdlICE9PSB1bmRlZmluZWQpXG4gICAgICAgIHJldHVybiBTd2FsLmZpcmUoe1xuICAgICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgICAgdGl0bGU6ICdPb3BzLi4uJyxcbiAgICAgICAgICB0ZXh0OiBgJHtyZXNwb25zZS5NZXNzYWdlfWBcbiAgICAgICAgfSk7XG4gICAgICBzZXRQb3N0aW5nKHBvc3RpbmdBbGwuZmlsdGVyKHggPT4geCAhPT0gcG9zdGluZykpO1xuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgIFN3YWwuZmlyZSh7XG4gICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgIHRleHQ6ICdJbnRlcm5hbCBFcnJvcidcbiAgICAgIH0pO1xuICAgIH1cbiAgfTtcblxuICByZXR1cm4gKFxuICAgIDw+XG4gICAgICA8YnV0dG9uXG4gICAgICAgIHR5cGU9XCJidXR0b25cIlxuICAgICAgICBvbkNsaWNrPXtvbkNsaWNrfVxuICAgICAgICBpZD1cImJ1dHRvblJlbW92ZVwiXG4gICAgICAgIGNzcz17W2J1dHRvbkNzc119XG4gICAgICA+XG4gICAgICAgIFJlbW92ZVxuICAgICAgPC9idXR0b24+XG4gICAgPC8+XG4gICk7XG59XG5cbmNvbnN0IGJ1dHRvbkNzcyA9IGNzc2BcbiAgYm9yZGVyOiAxcHggc29saWQgcmdiYSh2YXIoLS1kMGIsIDIxOSwgMjE5LCAyMTkpLCAxKTtcbiAgY29sb3I6IHJnYmEodmFyKC0tZjA3LCAzOCwgMzgsIDM4KSwgMSk7XG4gIGJhY2tncm91bmQtY29sb3I6IHdoaXRlO1xuICB3aWR0aDogMzAwcHg7XG4gIGhlaWdodDogNDhweDtcbiAgZm9udC1zaXplOiAxNHB4O1xuICBmb250LXdlaWdodDogYm9sZDtcbmA7XG5cbmNvbnN0IG1hcmdpbkxlZnQwID0gY3NzYFxuICBtYXJnaW4tbGVmdDogMHB4O1xuYDtcblxuLy8gUmVtb3ZlLnByb3BUeXBlcyA9IHtcbi8vICAgY3VycmVudFVzZXI6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgIHBvc3Rpbmc6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5vYmplY3QsIFByb3BUeXBlcy5hcnJheV0pLFxuLy8gICBzZXRDb21tZW50QVBJOiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGluZGV4T2ZDb21tZW50T25UaGlzUG9zdGluZzogUHJvcFR5cGVzLm51bWJlclxuLy8gfTtcblxuLy8gUmVtb3ZlLmRlZmF1bHRQcm9wcyA9IHtcbi8vICAgY3VycmVudFVzZXI6IFwiXCIsXG4vLyAgIHBvc3Rpbmc6IHt9LFxuLy8gICBpc0NvbW1lbnQ6IGZhbHNlXG4vLyB9O1xuXG5leHBvcnQgZGVmYXVsdCBSZW1vdmU7XG4iXX0= */",
        "toString": [Function],
      }
    }
    id="buttonRemove"
    onClick={[Function]}
    type="button"
  >
    <button
      className="css-e09scu-buttonCss-Remove"
      id="buttonRemove"
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
  </EmotionCssPropInternal>
</Remove>
`;
